src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/Contract.cs

```diff
@@ -6,11 +6,50 @@
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Runtime.CompilerServices;
+using System.Text;
 
 namespace Roslyn.Utilities
 {
     internal static class Contract
     {
+        [InterpolatedStringHandler]
+        private struct ThrowIfTrueInterpolatedStringHandler
+        {
+            private readonly StringBuilder? _stringBuilder;
+
+#pragma warning disable IDE0060 // Remove unused parameter - https://github.com/dotnet/roslyn/issues/58168
+            public ThrowIfTrueInterpolatedStringHandler(int literalLength, int formattedCount, bool condition)
+#pragma warning restore IDE0060 // Remove unused parameter
+            {
+                _stringBuilder = condition ? new StringBuilder(capacity: literalLength) : null;
+            }
+
+            public void AppendLiteral(string value) => _stringBuilder?.Append(value);
+
+            public void AppendFormatted<T>(T value) => _stringBuilder?.Append(value?.ToString());
+
+            public string GetFormattedText() => _stringBuilder?.ToString() ?? string.Empty;
```

Comment:
For these, I would actually assert that `_stringBuilder` isn't null. The API is being used wrong if it is.